### ISSUE TITLE
Mac's Text Replacement Doesn't Work

### ISSUE THREAD

**Issue by benawad** [ianstormtaylor/slate#3087]
#### Do you want to request a _feature_ or report a _bug_?

bug

#### What's the current behavior?

![slateg](https://user-images.githubusercontent.com/7872329/68037821-8baa5280-fc96-11e9-90d4-4fde1177c13b.gif)

#### What's the expected behavior?

`omw` should get deleted

**Comment by ryanmitts** [comment]
Maybe this comment will be helpful in the future. I tried this in my CR that has native browser manipulation on simple insertions (#3076) and it also breaks. I'm new to all the InputEvent stuff, but seems like the beforeinput event is the synthetic React one. It looks the same as an insertText event, so I don't know a way to know if it's replacing text without looking at the DOM change after allowing the native manipulation to proceed. Problem is, Chrome messes up and deletes a lot of the nested elements for the text Nodes, so it breaks the reconciliation.

I tried what happens in Chrome outside of React using the native beforeInput event which should support InputEvents level two, but it also reports as inputType of insertText. This doesn't make sense as the spec makes it seem like insertReplacementText is more appropriate. This is also how Safari reports the event. I raised a Chrome bug for for this in the meantime ([1021460](https://bugs.chromium.org/p/chromium/issues/detail?id=1021460#c4))
